### PR TITLE
Trigger PUD when RLS returns error and there are no images to display

### PR DESCRIPTION
**Pull request by olegrise**
## Description
- Fix for #37. 
- Add new error `"file-not-found"` log error specifically for `NOEXIST` RLS file state.
- Add `"file-insufficient-disk-space-error"` error for consistency with [rise-common-component implementation](https://github.com/Rise-Vision/rise-common-component/pull/17/files#diff-27a9ff70c8b416fab56632be0c1cb5f3R126).

## Motivation and Context
When Rise Local Storage returns an error for all images, then this is a the same case as there are no images in which case the `"report-done"` event is sent after the time configured in the `duration` attribute (see `_startEmptyPlayUntilDoneTimer()` method).

P.S. I created a [flowchart diagram](https://docs.google.com/drawings/d/1IvNnax3YJABrtIxNEJLfSdMABELEAvUvqEBgGEBqcuQ/edit?usp=sharing) for myself to better understand the `rise-image` logic and added it to the [rise-image Web Component Design](https://docs.google.com/document/d/1MtO6VGAyL2PxcX9qgTUEifueyf_nOBeqc6BAXeZtMF0/edit?usp=sharing) just for the reference. I decided to not include any of those docs into repo's readme because there are no plans in maintaining them. They are not shared publicly either. 

## How Has This Been Tested?
Tested manually. New automated tests are added to cover the fixed use case.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
NA

### CODE REVIEW

**Comment by olegrise** [comment]
@andrecardoso please review. 
@stulees @santiagonoguez FYI.

File: src/rise-image.js [hunk 1]
```diff
@@ -581,6 +583,17 @@ class RiseImage extends RiseElement {
         this._clearDisplayedImage();
       }
     }
+
+    //if all requested files have errors, then trigger PUD
+    if ( this._filesList ) {
+      let allFilesHaveErrors = this._filesList.every( filePath => {
```
Comment (andrecardoso) [inline src/rise-image.js]: I think this can be a `const`

Comment (olegrise) [reply src/rise-image.js]: @andrecardoso makes sense. Please review last commit. Thanks.

**Comment by olegrise** [comment]
https://trello.com/c/n7IEWilo/216-3-rise-image-37-done-event-is-not-sent-when-there-is-a-rise-local-storage-file-error

**Pull request merged**